File: viewer/src/main/java/com/tianxing/hotflyer/viewer/adapter/item/Linkable.java
package com.tianxing.hotflyer.viewer.adapter.item;

import java.io.Serializable;

import static com.tianxing.hotflyer.viewer.JAViewer.Objects_equals;

/**
 * Project: JAViewer
 */
public class Linkable implements Serializable {

    public String link;

    public String getLink() {
        return link;
    }

    @Override
    public boolean equals(Object linkable) {
        if (!(linkable instanceof Linkable)) {
            return false;
        }

        return Objects_equals(link, ((Linkable) linkable).link);
    }

    @Override
    public String toString() {
        return "Linkable{" +
                "link='" + link + '\'' +
                '}';
    }
}

File: viewer/src/main/java/com/tianxing/hotflyer/viewer/adapter/item/Linkable.kt
package com.tianxing.hotflyer.viewer.adapter.item

import com.tianxing.hotflyer.viewer.JAViewer.Companion.Objects_equals
import java.io.Serializable

/**
 * Project: JAViewer
 */
open class Linkable : Serializable {

    var link: String? = null

    override fun equals(linkable: Any?): Boolean {
        return if (linkable !is Linkable) {
            false
        } else Objects_equals(link, linkable.link!!)

    }

    override fun toString(): String {
        return "Linkable{" +
                "link='" + link + '\'' +
                '}'
    }
}
